web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,72 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
+The snapshot list is located under the VM. A tab item in the horizontal navigation will be dedicated for snapshots. The list is ordered by the creation date as a default. 
+The user can sort the list by name or date, clicking on the title will allow the user to switch sorting and decide if it's ascending or descending.  
+![Offline Snapshot - same modal as in VM list](img/2-01.jpg)
+Same modal as in VM list - Snapshot action
+![Offline Snapshot tab - expanded item](img/2-02.jpg)
+List item drawer. The user clicks any place on the row in order to expand the row for more information. 
+
+![Offline Snapshot tab - kebab actions](img/2-03.jpg)
+Kebab actions.
+Clicking on the kebab menu on each row will open an action menu with several options:
+- Create VM Template
+- Create VM
+- Restore
+- Delete
+![Offline Snapshot tab - restoring action](img/2-04.jpg)
+Restoring VM from snapshot modal. When clicking the restore option in the action menu it opens the restore popup. I tried keeping the restore popup similar to the cloning and template popups.
+
+
+## Scheduled snapshots
+
+This will be added in the future
+
+![Offline Schedule Snapshot - scheduled sub tab](img/3-00.jpg)
```

Comment:
"Shcedule" to "Schedule" in the blue button.

Should the "Day" column, which currently just says "30th" (which isn't very clear and could be next month's 30th), be switched to the same "Date" column of the regular Snapshots list? Maybe with the Time Zone included too?

	Date: Oct 12, 2018 9:35:51PM UTC+01:00

It's probably worth poking around other parts of the console to see what date format is used elsewhere. There's probably a standardized way of writing dates, maybe even with time zones included.

Reply:
I aligned it with this:
<img width="158" alt="Screen Shot 2019-06-16 at 9 29 29" src="https://user-images.githubusercontent.com/48942365/59562066-870b3780-9030-11e9-9fdd-0ed7d12d37ba.png">
But there is no mention to UTC. Does the system update the time according to the viewer?
Removing the UTC makes me question if we need this field at all.

Reply:
I suggested UTC/Time Zone because the original modal and Scheduled table of this PR did, but you're right that the rest of the console typically doesn't show that exact time information (unless you hover over a `Created At` field like the one in your image).

I believe that the times we show in the console are based on the cluster's time zone, not the viewer/user's. I can imagine this being confusing for admins who frequently manage cluster across multiple time zones, but that's an OpenShift-wide problem and probably out of scope for this PR.

So to align with the rest of OpenShift for now, I'd change my recommendation to:

```
Date: Oct 12, 9:35 pm
```

And if we wanted to explore the idea of combining Scheduled with Existing Snapshots (described above), then maybe we could clearly indicate when a Date hasn't happened yet with something like:

```
Date: Oct 12, 9:35 pm (scheduled)
```

This, along with the changes suggested previously, would probably make Scheduled snapshots stand out in a combined list.

> Removing the UTC makes me question if we need this field at all.

UTC aside, I'd say the new Date field of this Snapshots and Scheduled Snapshots list is probably one of the most important fields to include. Were you referring to the separate "Time Zone" column? I agree that can be removed.

Reply:
@andybraren I updated the MD to combined list